src/components/dropdownSearch/DropdownSearch.tsx

```diff
@@ -0,0 +1,314 @@
+    highlightThreshold: 100,
+    highlightAllFilterResult: false,
+    noResultText: 'No results',
+  };
+
+  private getSelectedOption(): JSX.Element {
+    return <span key={this.props.selectedOption.value}
+                 className='dropdown-selected-value'
+                 data-value={this.props.selectedOption.value}>
+      {this.props.selectedOption.displayValue || this.props.selectedOption.value}
+    </span>;
+  }
+
+  private getDropdownOptions(): JSX.Element[] {
+    const options = _.chain(this.props.options)
+      .filter(
+        (option: IDropdownOption) => {
+          const value = option.displayValue || option.value;
+
+          return _.isEmpty(this.props.filterText) || s.contains(value.toLowerCase(), this.props.filterText.toLowerCase());
+        },
+      )
+      .map((option: IDropdownOption, index, options: IDropdownOption[]) => {
+        const optionClasses: string = option.value === this.props.selectedOption.value ? 'state-selected' : '';
+        const value = option.displayValue || option.value;
+        const valueToShow = !!this.props.highlightAllFilterResult || options.length <= this.props.highlightThreshold
+          ? this.getTextFiltered(value)
+          : value;
+        return <li key={option.value}
+                   className={option === this.props.activeOption ? 'active' : ''}>
+          <span className={optionClasses}
+                onMouseDown={(e: React.MouseEvent<HTMLSpanElement>) => this.handleOnOptionClick(e)} data-value={option.value}>
+            {this.getDropdownPrepend(option)}
+            {this.getSvg(option)}
+            {valueToShow}
+          </span>
+        </li>;
+      })
+      .value();
+
+    if (!options.length) {
+      return [
+        <li key='noResultDropdownSearch'>
+          <span>{this.props.noResultText}</span>
+        </li>,
+      ];
+    }
+
+    return options;
+  }
+
+  private getSvg(option: IDropdownOption): JSX.Element {
+    if (option.svg) {
+      return <span key={option.svg.name}
+                   className='value-icon'>
+        <Svg {...option.svg} />
+      </span>;
+    }
+    return null;
+  }
+
+  private getTextFiltered(text: string): (JSX.Element | string)[] | string {
+    this.textIndex = 0;
+    const originalText = new String(text).toString();
+    if (!_.isEmpty(this.props.filterText)) {
+      const textFilterElements: (JSX.Element | string)[] = [''];
+      let index: number = text.toLowerCase().indexOf(this.props.filterText.toLowerCase());
+      while (index !== -1) {
+        if (index > 0) {
+          textFilterElements.push(text.substring(0, index));
+        }
+        textFilterElements.push(this.getTextElement(text.substring(index, index + this.props.filterText.length), originalText, 'bold'));
+        text = text.substring(index + this.props.filterText.length);
+        index = text.toLowerCase().indexOf(this.props.filterText.toLowerCase());
+      }
+      textFilterElements.push(text);
+
+      this.textIndex = 0;
```

Comment:
It does not seem like you have changed this value since you've set it to 0 at the start of your function. Is this needed?